### ISSUE TITLE
All devices unverified

### ISSUE THREAD

**Issue by derekoharrow** [gcobb321/icloud3#306]
Since rc10.3 (I think) all my icloud devices are showing as unverified and are not being tracked.

There are the following errors in the HA log file:

```
2024-02-19 11:30:47.363 ERROR (SyncWorker_31) [icloud3] list index out of range
Traceback (most recent call last):
  File "/config/custom_components/icloud3/support/start_ic3_control.py", line 208, in stage_4_setup_data_sources
    if start_ic3.setup_tracked_devices_for_famshr() is False:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/icloud3/support/start_ic3.py", line 1358, in setup_tracked_devices_for_famshr
    _check_duplicate_device_names(PyiCloud, _FamShr)
  File "/config/custom_components/icloud3/support/start_ic3.py", line 1625, in _check_duplicate_device_names
    famshr_fnames_base  = [_fname_base(famshr_fname)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/icloud3/support/start_ic3.py", line 1731, in _fname_base
    suffix = fname.split('(')[1].split(')')[0]
             ~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-02-19 11:30:47.384 ERROR (SyncWorker_31) [icloud3] list index out of range
Traceback (most recent call last):
  File "/config/custom_components/icloud3/support/start_ic3_control.py", line 208, in stage_4_setup_data_sources
    if start_ic3.setup_tracked_devices_for_famshr() is False:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/icloud3/support/start_ic3.py", line 1358, in setup_tracked_devices_for_famshr
    _check_duplicate_device_names(PyiCloud, _FamShr)
  File "/config/custom_components/icloud3/support/start_ic3.py", line 1625, in _check_duplicate_device_names
    famshr_fnames_base  = [_fname_base(famshr_fname)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/icloud3/support/start_ic3.py", line 1731, in _fname_base
    suffix = fname.split('(')[1].split(')')[0]
             ~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-02-19 11:30:48.006 ERROR (SyncWorker_31) [icloud3] list index out of range
Traceback (most recent call last):
  File "/config/custom_components/icloud3/support/start_ic3_control.py", line 208, in stage_4_setup_data_sources
    if start_ic3.setup_tracked_devices_for_famshr() is False:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/icloud3/support/start_ic3.py", line 1358, in setup_tracked_devices_for_famshr
    _check_duplicate_device_names(PyiCloud, _FamShr)
  File "/config/custom_components/icloud3/support/start_ic3.py", line 1625, in _check_duplicate_device_names
    famshr_fnames_base  = [_fname_base(famshr_fname)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/icloud3/support/start_ic3.py", line 1731, in _fname_base
    suffix = fname.split('(')[1].split(')')[0]
             ~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-02-19 11:30:48.160 ERROR (SyncWorker_31) [icloud3] list index out of range
Traceback (most recent call last):
  File "/config/custom_components/icloud3/support/start_ic3_control.py", line 208, in stage_4_setup_data_sources
    if start_ic3.setup_tracked_devices_for_famshr() is False:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/icloud3/support/start_ic3.py", line 1358, in setup_tracked_devices_for_famshr
    _check_duplicate_device_names(PyiCloud, _FamShr)
  File "/config/custom_components/icloud3/support/start_ic3.py", line 1625, in _check_duplicate_device_names
    famshr_fnames_base  = [_fname_base(famshr_fname)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/icloud3/support/start_ic3.py", line 1731, in _fname_base
    suffix = fname.split('(')[1].split(')')[0]
             ~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
```

Also, if I go into config settings -> cloud account and mobile app it says:

`The Mobile App Integration is not installed. The Mobile App will not be used as a data source; location data and zone enter/exit triggers will not be monitored`

But, the Mobile App integration is installed and working just fine.

In Stage 5 of the log file, it has Red Crosses next to each device saying NOT Tracked Device and a red cross next to Mobile App Tracking Parameters, and Notifications says Waiting for Notify Svc to Start.

**Comment by gcobb321** [comment]
I just uploaded rc10.4 yesterday afternoon. Download that one and lets see if this happens. If so, I'll need your icloud3-0.log file to see what is going on. Also export the event log and send me that. I did some work on these log files to make them a lot easier to see what was going on.

**Comment by gcobb321** [comment]
I added error catching code to the error you encountered above (list index out of range) to prevent that error. It probably has to do with the Mobile App Integration not being set up. Do you see the Mobile App Integration on the Integration page with the iPhone and other iDevices listed?

Unzip start_ic3.ip into the icloud3/support directory and restart ha
[start_ic3.zip](https://github.com/gcobb321/icloud3/files/14335841/start_ic3.zip)

**Comment by executed** [comment]
what was the solution here?

**Comment by gcobb321** [comment]
@executed I don't understand the question. That issue was posted in February and dealt with a problem in a beta version of iCloud3. This is now August and the latest release is v3.0.5.7. Get that through HACS or on the iCloud3 GitHub repository.

**Comment by executed** [comment]
@gcobb321 I just had an "Unknown error" when migrating from v2 and trying to add the same devices I had before.
I think the error was caused by trying to use the same device IDs I used before for iCloud integration.

**It all started working** after I slightly adjusted IDs, but hey, I wasn't looking at the logs at the time, I should say logs are good Wizard is cool too, and I don't need to store my pwd in plain text!!
Thank you for your **excellent** work over the years, Gary!
